packages/web2/app/actions/domain.ts

```diff
@@ -12,3 +20,96 @@ export const getDomains = createServerFn({
     assertStatus(domains, 200);
     return domains.data;
   });
+
+export const getPaths = createServerFn({
+  method: "GET",
+})
+  .validator(z.string())
+  .handler(async ({ data: domain }) => {
+    if (!domain) return null;
+    try {
+      const response = await fetch(`https://${domain}/sitemap.xml`);
+      if (!response.ok) return null;
+      const text = await response.text();
+      const parser = new XMLParser();
+      const xml = parser.parse(text);
+
+      const countPathSegments = (url: string): number => {
+        const path = extractPathFromUrl(url);
+        return path.split("/").filter(Boolean).length;
+      };
+
+      const locations: string[] = xml.urlset.url
+        .map(({ loc }: { loc: string }) => loc)
+        .filter((url: string) => {
+          const segmentCount = countPathSegments(url);
+          return segmentCount <= 1;
+        });
+
+      return locations;
+    } catch (e) {
+      console.error(e);
+      return null;
+    }
+  });
+
+function extractLinks($: CheerioAPI, el: BasicAcceptedElems<AnyNode>) {
+  return $(el)
+    .find("a")
+    .map((_, a) => {
+      const href = $(a).attr("href");
+      return href ? resolveUrl(href) : null;
+    })
+    .get()
+    .filter(Boolean);
+}
+
+function resolveUrl(relative: string) {
+  if (relative.startsWith("/") && !relative.startsWith("/#")) {
+    return relative;
+  }
+  return null;
+}
+
+export const getPathsFromMenu = createServerFn({
+  method: "GET",
+})
+  .validator(z.string())
+  .handler(async ({ data: domain }) => {
+    if (!domain) return null;
+    try {
+      const response = await axios.get("https://" + domain);
+      const $ = cheerio.load(response.data);
+
+      const navLinks: string[] = [];
+
+      // === STRATEGY 1: <nav> elements ===
+      $("nav").each((_, nav) => {
+        const links = extractLinks($, nav);
+        navLinks.push(...links);
+      });
+
+      // === STRATEGY 2: [role="navigation"] ===
+      if (navLinks.length === 0) {
+        $('[role="navigation"]').each((_, nav) => {
+          const links = extractLinks($, nav);
+          navLinks.push(...links);
+        });
+      }
+
+      // === STRATEGY 3: <ul> or <div> with at least 3 <a> ===
+      if (navLinks.length === 0) {
+        $("ul, div").each((_, el) => {
+          const links = extractLinks($, el);
+          navLinks.push(...links);
+        });
+      }
+
+      return navLinks;
+    } catch (error: unknown) {
+      setResponseStatus(412);
+      if (error instanceof Error) {
+        throw new Error(error?.message ?? "Error reading sitemap");
```

Comment:
Hier wird keine sitemap geladen. 


```suggestion
        throw new Error(error?.message ?? "Error reading links from page");
```